### ISSUE TITLE
scribusUnstable not building

### ISSUE THREAD

**Issue by schlichtanders** [NixOS/nixpkgs#148139]
### Describe the bug
Just switched to channel 21.11 however I had to disable `scribusUnstable` for it to work.

### Steps To Reproduce
1. switch to 21.11
2. install `pkgs.scribusUnstable`
3. view error during build

### Expected behavior
scribusUnstable should build without any problems

### Additional context
here some output from the build
```
/build/scribus-1.5.7/scribus/fonts/sfnt.cpp: In function 'QByteArray sfnt::subsetFaceWithHB(const QByteArray&, QList<unsigned int>, int, QMap<unsigned int, unsigned int>&)':
/build/scribus-1.5.7/scribus/fonts/sfnt.cpp:1260:3: error: 'hb_subset_input_set_retain_gids' was not declared in this scope; did you mean 'hb_subset_input_set_flags'?
1260 |   hb_subset_input_set_retain_gids(hbSubsetInput.get(), true);
|   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
|   hb_subset_input_set_flags
/build/scribus-1.5.7/scribus/fonts/sfnt.cpp:1261:3: error: 'hb_subset_input_set_drop_hints' was not declared in this scope;did you mean 'hb_subset_input_set_user_data'?
1261 |   hb_subset_input_set_drop_hints(hbSubsetInput.get(), false);
|   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
|   hb_subset_input_set_user_data
/build/scribus-1.5.7/scribus/fonts/sfnt.cpp:1263:3: error: 'hb_subset_input_set_name_legacy' was not declared in this scope; did you mean 'hb_subset_input_set_flags'?
1263 |   hb_subset_input_set_name_legacy(hbSubsetInput.get(), true);
|   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
|   hb_subset_input_set_flags
/build/scribus-1.5.7/scribus/fonts/sfnt.cpp:1266:57: error: 'hb_subset' was not declared in this scope; did you mean 'hb_set_set'?
1266 |   QScopedPointer<hb_face_t, HbFaceDeleter> hbSubsetFace(hb_subset(hbFullFace.get(), hbSubsetInput.get()));
|                                                         ^~~~~~~~~
|                                                         hb_set_set
make[2]: *** [scribus/CMakeFiles/scribus.dir/build.make:7199: scribus/CMakeFiles/scribus.dir/fonts/sfnt.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2177: scribus/CMakeFiles/scribus.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
builder for '/nix/store/fj357q3nxqkz3yscrkr73vzxvy0h9x9m-scribus-1.5.7.drv' failed with exit code 2
cannot build derivation '/nix/store/lfxxd42ip5g9p7jh1wi8spsdh5kg70ji-system-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/l5g12ghfbkfgrrhdyp1gdsk2slgli7c1-nixos-system-gram17-21.11.333823.96b4157790f.drv': 1 dependencies couldn't be built
error: build of '/nix/store/l5g12ghfbkfgrrhdyp1gdsk2slgli7c1-nixos-system-gram17-21.11.333823.96b4157790f.drv' failed
```

### Notify maintainers
@erictapen @Kiwi 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
- host os: `Linux 5.14.10, NixOS, 21.11 (Porcupine)`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.3.16`
- channels(root): `"nixos-21.11.333823.96b4157790f"`
- nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by erictapen** [comment]
This was fixed in 35d624300b259b5508eea97ed346599f38bcfee5. Unfortunately the backport came a bit close to the release, so you might need to wait a few days until it hits the `nixos-21.11` channel.

**Comment by erictapen** [comment]
It just got published, so updating your `nixos-21.11` channel should produce a building `scribusUnstable` from now on.

**Comment by schlichtanders** [comment]
worked - awesome!
